ambient_toolbox/utils/model.py

```diff
@@ -29,3 +29,41 @@ def object_to_dict(obj, blacklisted_fields: list = None, include_id: bool = Fals
             valid_data[key] = value
 
     return valid_data
+
+
+def get_cached_related_obj(obj: Model, related_field_name: str, silently_return_none: bool = False):
+    """
+    This function helps to avoid silent sub-queries, due to missing `select_related()` or `prefetch_related()`.
+
+    This function performs a lookup in the fields_cache of the given object.
+    If silently_return_none is True, this function will return None if the field is not cached.
+    Otherwise, an AttributeError will be raised.
+
+    What problem does this function solve?
+
+    foo = Foo.objects.all().get()
+    foo.bar  # <-- this will silently cause a sub-query
+
+    foo = Foo.objects.all().select_related('bar').first()
+    foo.bar # <-- this will cause no sub-query
+
+    The only difference is the usage of `select_related()`, which can easily be forgotten.
+
+    Usage:
+    get_cached_related_obj(foo, 'bar')
+
+    This will result in a KeyError instead of a silent sub-query if the field is not cached
+    via `select_related()` or `prefetch_related()`.
+    """
+    fields_cache = obj._state.fields_cache  # warning: this is an undocumented feature of django
```

Comment:
nitpick: i prefer inline comments to be above the LOC

Reply:
not sure if that makes it more readable, but if makes you happy ... 🫶